### PR TITLE
Add Vulkan-ExtensionLayer

### PR DESCRIPTION
**Pull request by dneto0**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -174,3 +180,8 @@ endif()
 if (${VKB_BUILD_VULKAN_VALIDATION_LAYERS})
   include(cmake/vulkan-validation-layers.cmake)
 endif()
+
+# Vulkan-ExtensionLayer requires Vulkan-Headers
```
Comment (dj2) [inline CMakeLists.txt]: I don't think this is necessary, most of the things in the Vulkan section require headers to come first.

File: cmake/vulkan-extensionlayer.cmake [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+message(STATUS "VKB Vulkan-ExtensionLayer: ${VKB_VULKAN_EXTENSION_LAYER_SOURCE_DIR}")
+
+if(${CMAKE_SYSTEM_NAME} MATCHES "Android")
+  message(STATUS "VKB Vulkan-ExtensionLayer: Skipping on Android")
+else()
+  if(MSVC)
+    set(BUILD_TESTS OFF CACHE BOOL "Vulkan-ExtensionLayer tests off")
+    message(STATUS "VKB Vulkan-ExtensionLayer: Disabling tests")
+  else()
+    set(BUILD_TESTS ON CACHE BOOL "Vulkan-ExtensionLayer tests on")
```
Comment (dj2) [inline cmake/vulkan-extensionlayer.cmake]: nit: /Vulkan-ExtensionLayer tests on/Vulkan-ExtensionLayer: Enabling tests

**Review by dj2 — COMMENTED** [review]
CMake Error at third_party/vulkan-extensionlayer/layers/CMakeLists.txt:20 (add_custom_target):
  add_custom_target cannot create target "mk_layer_config_dir" because
  another target with the same name already exists.  The existing target is a
  custom target created in source directory
  "T:/src/github/vkb/third_party/vulkan-validationlayers/layers".  See

**Comment by dneto0** [comment]
Expected to fail.

**Comment by dneto0** [comment]
Failure is getting pthread.h on Windows.

**Comment by dj2** [comment]
I think you're either going to have to port Vulkan-ExtensionLayer to windows, or disable it in VKB on windows ....

**Comment by dneto0** [comment]
I decided to disable it on Windows.
I also rolled forward to latest upstream Vulkan-ExtensionLayer

**Pull request merged**